### PR TITLE
docs: Fix some typos

### PR DESCRIPTION
**Pull request by janosh**
started reading the docs today to learn some polars and noticed couple typos

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
> started reading the docs today to learn some polars

legend, mate

looks good (I had to check whether "re-use" was correct, and to be fair it doesn't appear in https://www.merriam-webster.com/dictionary/)

File: py-polars/polars/io/database.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def __exit__(
         exc_val: BaseException | None,
         exc_tb: TracebackType | None,
     ) -> None:
-        # iif we created it and are finished with it, we can
+        # if we created it and are finished with it, we can
```
Comment (stinodego) [inline py-polars/polars/io/database.py:154]: iif is actually a valid term and I think it applies here:
https://en.wikipedia.org/wiki/If_and_only_if

Could you revert this one?

**Review by stinodego — CHANGES_REQUESTED** [review]
Thanks! One comment then this can be merged.

File: py-polars/polars/io/database.py (hunk 1 above)
Comment (janosh) [reply py-polars/polars/io/database.py:154]: that's briefly what i thought too but then it should be iff

Comment (stinodego) [reply py-polars/polars/io/database.py:154]: oh you're right.

**Review by stinodego — APPROVED** [review]
:tada:

**Pull request merged**